### ISSUE TITLE
SyntaxError after autocorrecting long line with a block and appended string 

### ISSUE THREAD

**Issue by tom-kuca** [rubocop/rubocop#11663]
Given the input file

```ruby
Logger.error "\n\n\nBacktrace:\n\s\s" + e.backtrace.select { |line| line.start_with?(LIBRARY_PATH) }.join("\n\s\s") + "\n\n"

```

the command 

```
exe/rubocop --config /dev/null --only Style/BlockDelimiters,Layout/LineLength task.rb -a
```

autocorrects the file to 

```ruby
Logger.error "\n\n\nBacktrace:\n\s\s" + e.backtrace.select do |line|
 line.start_with?(LIBRARY_PATH) end.join("\n\s\s") + "\n\n"
```

There is a syntax error:
```
$ ruby task.rb                                                                            
/tmp/task.rb:2: syntax error, unexpected '+', expecting end-of-input
...RARY_PATH) end.join("\n\s\s") + "\n\n"
```
No error is reported on the original file.

### Expected behavior 

Expected behavior is a syntactically correct ruby file after autocorrect. 

### Version

```
❯ exe/rubocop -V --config /dev/null
1.47.0 (using Parser 3.2.1.0, rubocop-ast 1.27.0, running on ruby 3.0.2) [x86_64-linux]
```
